jupyterlab_bigquery/jupyterlab_bigquery/tests/pagedAPI_handler/pagedQuery_handler_test.py

```diff
@@ -0,0 +1,247 @@
+      'pagedQuery_handler.PagedQueryHandler.client.query'
+  )
+  @patch.object(bigquery, 'QueryJobConfig')
+  def test_query_dry(self, fake_job_config, fake_client_query):
+    query = MagicMock()
+    jobConfig = MagicMock()
+    dryRunOnly = True
+    page_size = 2000
+    job_id = 1231
+
+    request_body = {
+        'query': query,
+        'jobConfig': jobConfig,
+        'dryRunOnly': dryRunOnly
+    }
+
+    dry_run_job = MagicMock()
+    dry_run_job.job_id = job_id
+
+    dry_run_job_config_mock = MagicMock()
+
+    fake_client_query.return_value = dry_run_job
+    fake_job_config.return_value = dry_run_job_config_mock
+
+    gen = self.dummy_query_handler.query(request_body, page_size)
+
+    actual_dry_run_job, actual_job_id = next(gen)
+
+    fake_job_config.assert_called_with(dry_run=True, use_query_cache=False)
+    fake_client_query.assert_called_with(query,
+                                         job_config=dry_run_job_config_mock)
+    self.assertEqual(actual_dry_run_job, dry_run_job)
+    self.assertEqual(actual_job_id, job_id)
+
+    with self.assertRaises(StopIteration):
+      next(gen)
+
+
+  @patch(
+      'jupyterlab_bigquery.pagedAPI_handler.'+\
+      'pagedQuery_handler.PagedQueryHandler.client.query'
+  )
+  @patch.object(bigquery, 'QueryJobConfig')
+  def test_query_dry_general_err(self, fake_job_config, fake_client_query):
+    query = MagicMock()
+    jobConfig = MagicMock()
+    dryRunOnly = True
+    page_size = 2000
+    job_id = 1231
+
+    request_body = {
+        'query': query,
+        'jobConfig': jobConfig,
+        'dryRunOnly': dryRunOnly
+    }
+
+    dry_run_job = MagicMock()
+    dry_run_job.job_id = job_id
+
+    dry_run_job_config_mock = MagicMock()
+
+    fake_client_query.side_effect = Exception()
+    fake_job_config.return_value = dry_run_job_config_mock
+
+    gen = self.dummy_query_handler.query(request_body, page_size)
+
+    with self.assertRaises(Exception):
+      next(gen)
+
+  @patch(
+      'jupyterlab_bigquery.pagedAPI_handler.'+\
+      'pagedQuery_handler.PagedQueryHandler.client.query'
+  )
+  @patch.object(bigquery, 'QueryJobConfig')
+  def test_query_dry_erres_err(self, fake_job_config, fake_client_query):
+    query = MagicMock()
+    jobConfig = MagicMock()
+    dryRunOnly = True
+    page_size = 2000
+    job_id = 1231
+    dummy_err_msg = 'dummy_err_msg'
+
+    request_body = {
+        'query': query,
+        'jobConfig': jobConfig,
+        'dryRunOnly': dryRunOnly
+    }
+
+    dry_run_job = MagicMock()
+    dry_run_job.job_id = job_id
+
+    dry_run_job_config_mock = MagicMock()
+
+    class DummyException(Exception):
+      errors = [{'message': dummy_err_msg}]
+
+    err = DummyException()
+
+    fake_client_query.side_effect = err
+    fake_job_config.return_value = dry_run_job_config_mock
+
+    gen = self.dummy_query_handler.query(request_body, page_size)
+
+    try:
+      next(gen)
+      self.fail()
+    except Exception as inst:
```

Comment:
Use `assertRaises` or `assertRaisesRegex` here and below instead:

https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaisesRegex